tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -117,32 +146,28 @@ ZTEST_USER(uart_async_single_read, test_single_read)
 	zassert_equal(k_sem_take(&rx_buf_released, K_MSEC(100)),
 		      0,
 		      "RX_BUF_RELEASED timeout");
+	uart_rx_disable(uart_dev);
+
 	zassert_equal(k_sem_take(&rx_disabled, K_MSEC(1000)), 0,
 		      "RX_DISABLED timeout");
 	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(100)), -EAGAIN,
 		      "Extra RX_RDY received");
 
-	zassert_equal(memcmp(tx_buf, rx_buf+5, 5), 0, "Buffers not equal");
-	zassert_equal(tx_aborted_count, 0, "TX aborted triggered");
+	zassert_equal(memcmp(tx_buf, tdata.last_rx_buf, 5), 0, "Buffers not equal");
```

Comment:
How do you guarantee that tdata.last_rx_buf is not pointing somewhere in rx_buf other than &rx_buf[5] if the driver does not release the buffer after the first 5 bytes?
The success/failure now depends on the timeout in uart_rx_enable().